File: solutions/python3/1015.py
class Solution:
    def smallestRepunitDivByK(self, K: int) -> int:
        if K % 10 not in {1, 3, 7, 9}:
            return -1

        set_ = set()
        mod = 0

        for N in range(1, K + 1):
            mod = (mod * 10 + 1) % K
            if mod == 0:
                return N
            if mod in set_:
                return -1
            set_.add(mod)

        return -1

File: solutions/java/1015.java
class Solution {
  public int smallestRepunitDivByK(int K) {
    if (K % 10 != 1 && K % 10 != 3 && K % 10 != 7 && K % 10 != 9)
      return -1;

    Set<Integer> set = new HashSet<>();
    int mod = 0;

    for (int N = 1; N <= K; ++N) {
      mod = (mod * 10 + 1) % K;
      if (mod == 0)
        return N;
      if (set.contains(mod))
        return -1;
      set.add(mod);
    }

    return -1;
  }
}